### ISSUE TITLE
Using print representation of symbols

### ISSUE THREAD

**Issue by Yehouda** [defaultxr/cl-patterns#37]
In patterns.lisp it generates the pstream class twice using **mutility:concat**:
https://github.com/defaultxr/cl-patterns/blob/a3139ac3591afc12334d29e032db985442e4d541/src/patterns/patterns.lisp#L612
https://github.com/defaultxr/cl-patterns/blob/a3139ac3591afc12334d29e032db985442e4d541/src/patterns/patterns.lisp#L891

That is a problem because **mutility:concat** prints it argument using ~a, and for symbols that depends on the value of ***print-case***. One of our customers who sets ***print-case*** to **:downcase** got an error because it was using **pbind-PSTREAM** instead **PBIND-PSTREAM**.

The proper fix is to have a function that generates the PSTREAM clas in a way that is independnet of global settings, e.g.:
```
(defun class-symbol-name-to-stream-class-name (class-symbol-name)
  (intern (concatenate 
           'string 
           (symbol-name class-symbol-name); make sure that symbol name not afected by *print-case*
           "-PSTREAM")
          (symbol-package class-symbol-name)))
```
And then use it in all cases when the class name is generated by code,. The places that I can see are the two lines above and:
https://github.com/defaultxr/cl-patterns/blob/a3139ac3591afc12334d29e032db985442e4d541/src/patterns/patterns.lisp#L30
https://github.com/defaultxr/cl-patterns/blob/a3139ac3591afc12334d29e032db985442e4d541/src/patterns/patterns.lisp#L33
https://github.com/defaultxr/cl-patterns/blob/a3139ac3591afc12334d29e032db985442e4d541/src/patterns/export.lisp#L7

**Comment by defaultxr** [comment]
Thanks, should be fixed now.

**Comment by Yehouda** [comment]
You missed this one:

https://github.com/defaultxr/cl-patterns/blob/c6c4975cd5cec14ccd4334cd6e6f48172b183c64/src/patterns/patterns.lisp#L33

**Comment by defaultxr** [comment]
I may be misunderstanding something, but it seems like `symbolicate` is not dependent on the value of `*print-case*`:

```lisp
(let ((*print-case* :downcase))
  (symbolicate 'foo '-pstream)) ; results in FOO-PSTREAM
```

**Comment by Yehouda** [comment]
Yes, but for consistency it always better, in situtaion when you make the same symbol (in general,  the same object) in more than one place, to use exactly the same code. 

Also makes it easier to understand the code: `(pattern-pstream-class-name pattern)` tells you what it tries to achieve, while `(symbolicate pattern '-pstream)` doesn't, and makes it easier to find all the occurrences when these symbols are made *by searching for *8pattern-pstream-class-name**).

**Comment by defaultxr** [comment]
Yes, you're right, that makes sense. I went and fixed that last one in 35ffbc3e5010984a08a3478f823dd21d31c74c8b. Thanks again.
